### ISSUE TITLE
Use the .Net Core RegAllocator for specific use cases

### ISSUE THREAD

**Issue by ciplogic** [dotnet/coreclr#8948]
I've noticed looking into RyuJIT code that .Net uses a LSRA algorithm and it is all fine. But for some use-cases a full register allocator (which I understand is supported by .Net Core) could be really nice.

Support a way to use the .Net Core's  (maybe it is already possible, yet I have no clue how) code gen for a specific method and allow a way of calling with minimal overhead between these two worlds.

I think as use-cases: games, use cases when JIT is to expensive to be called, maybe some math patterns where .Net Core generates a better code.

The bug is fuzzier, as I think that I don't want to offer a solution with the problem, but mostly to argue with the idea that either there should be an integration of the C2 compiler (or LLVM, or who knows which backend code generator you have with .Net Core) with RyuJit in a supported way and on both ends to be used .Net great toolset (C++ CLI is really not the best way to get good code, we love C# !).

**Comment by mikedn** [comment]
>  But for some use-cases a full register allocator

What's that? It's not like LSRA is half of a register allocator.

> which I understand is supported by .Net Core

Not sure what you mean by supported and what that has to do with .NET Core. What's for sure is that RyuJIT has a single register allocator - LSRA - and that both .NET Framework and .NET Core use the same JIT (on x64 at least) and thus the same register allocator.

> there should be an integration of the C2 compiler (or LLVM, or who knows which backend code generator you have with .Net Core)

There's https://github.com/dotnet/llilc which uses LLVM. However, the development of that is on hold for now, it's not that easy to get LLVM to work well "with managed" code. There's also .NET Native which uses C2, or so they claim...

**Comment by ciplogic** [comment]
Thank you @mikedn !
I am surprised, but you are right and thank you for clarification. I would assume that .Net Core did what I confused with .Net Core.

I opened this bug for a pragmatic reason: I wanted to make a kind of simple OpenGL game and I noticed that the bindings are really ... (I would use a bad word here) but on the plain .Net side (or Mono, and so on) there are many wrappers (like OpenTK, and so on). And I could see that some game developers would want a more game-friendly runtime as it should be a C++ like optimized and I assumed that .Net Core is this runtime.

I can see many cases when someone would use this approach to write something in C# still and it would be really nice to be inside the same process.

As my understanding now that .Net Core does much less things than .Net Native or LLILC, it may happen that this „proposal” could be a bit off and the issue could be closed.
